### PR TITLE
adde higher order array methods 

### PR DESCRIPTION
**Pull request by Sloth-Panda**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description

(Write your answer here.)
Added higher order array methods
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#2059

### CODE REVIEW

**Comment by Sloth-Panda** [comment]
I am  a part of DWOC @HarshCasper

**Comment by Sloth-Panda** [comment]
can you please also add level to it @atarax665 @HarshCasper

File: JavaScript/other/filter.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//(Filter)
+//The filter() method takes each element in an array and it applies a conditional statement against it.If this conditional returns true, the element gets pushed to the output array. If the condition returns false, the element does not get pushed to the output array.
```
Comment (atarax665) [inline JavaScript/other/filter.js:2]: Break such long comments in small multi-line comments.

File: JavaScript/other/filter.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+//Example-1
+const numbers = [1, 2, 3, 4];
+const evens = numbers.filter(item => item % 2 === 0);
+console.log(evens); // [2, 4]
```
Comment (atarax665) [inline JavaScript/other/filter.js:7]: Shift the output below the code.

File: JavaScript/other/map.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+
```
Comment (atarax665) [inline JavaScript/other/map.js:1]: Remove this blank line.

File: JavaScript/other/map.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+//(Map)
+//It will basically create a new array from an existing array applying the method to each member of the array
+
+const numbers = [1, 2, 3, 4];
+const doubled = numbers.map(item => item * 2);
+console.log(doubled); // [2, 4, 6, 8]
```
Comment (atarax665) [inline JavaScript/other/map.js:7]: Shift the output below this line.

File: JavaScript/other/map.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
```
Comment (atarax665) [inline JavaScript/other/map.js:8]: Remove this extra blank line.

File: JavaScript/other/map.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+const numbers = [1, 2, 3, 4];
+const sub = numbers.filter(item => item - 2);
+console.log(sub); // [-1, 0, 1, 2]
```
Comment (atarax665) [inline JavaScript/other/map.js:12]: Shift the output below the code and add a blank line at the end of file.

File: JavaScript/other/sort.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+// sort() as the name suggests basically sorts in ascending or descending order
+let numbers = [15, 2, 50, 55, 90, 5, 4, 9, 10];
+console.log(numbers.sort((num1, num2) => num1 - num2)); // Sorting numbers in accending order => result is [2, 4, 5, 9, 10, 15, 50, 55, 90]
+
+
+// Chaining all above four functions(Map, Sort, Filter and Reduce funtions)
+let numbers = [15, 2, 50, 55, 90, 5, 4, 9, 10];
+let initialValue = 5;
+const result = numbers
+    .map(number => number * 2)  // [30, 4, 100, 110, 180, 10, 8, 18, 20]//
```
Comment (atarax665) [inline JavaScript/other/sort.js:11]: Shift the output below the respective lines of code.

File: JavaScript/other/sort.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+// Chaining all above four functions(Map, Sort, Filter and Reduce funtions)
+let numbers = [15, 2, 50, 55, 90, 5, 4, 9, 10];
+let initialValue = 5;
+const result = numbers
+    .map(number => number * 2)  // [30, 4, 100, 110, 180, 10, 8, 18, 20]// 
+    .sort((num1, num2) => num1 - num2) // [4, 8, 10, 18, 20, 30, 100, 110, 180]
+    .filter(number => number > 20) // [30, 100, 110, 180]
+    .reduce((total, number) => total + number, initialValue); //425
+
+console.log(result); // Answer is 425
```
Comment (atarax665) [inline JavaScript/other/sort.js:16]: Add a blank line at the end of the file.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Sloth-Panda** [comment]
https://github.com/TesseractCoding/NeoAlgo/pull/2112/commits/2fa7a1610cc5d5e4b708ba77443f91b0ba81cfc5

@atarax665  I have updated the file this is actually the updated commit mistakenly pushed to master from a different branch take a look

**Comment by Sloth-Panda** [comment]
@HarshCasper @atarax665 can anyone check this its been 9-10 days now

File: JavaScript/other/filter.js (hunk 1 above)
Comment (HarshCasper) [reply JavaScript/other/filter.js:2]: It has not been resolved

File: JavaScript/other/sort.js (hunk 8 above)
Comment (HarshCasper) [reply JavaScript/other/sort.js:16]: Not done yet

File: JavaScript/other/filter.js (hunk 2 above)
Comment (HarshCasper) [reply JavaScript/other/filter.js:7]: Not done yet

File: JavaScript/other/map.js (hunk 3 above)
Comment (HarshCasper) [reply JavaScript/other/map.js:1]: Not done yet

File: JavaScript/other/map.js (hunk 4 above)
Comment (HarshCasper) [reply JavaScript/other/map.js:7]: Not done yet

File: JavaScript/other/map.js (hunk 5 above)
Comment (HarshCasper) [reply JavaScript/other/map.js:8]: Not done yet

File: JavaScript/other/map.js (hunk 6 above)
Comment (HarshCasper) [reply JavaScript/other/map.js:12]: Not done yet

File: JavaScript/other/sort.js (hunk 7 above)
Comment (HarshCasper) [reply JavaScript/other/sort.js:11]: Not done yet

File: JavaScript/other/filter.js (hunk 1 above)
Comment (HarshCasper) [reply JavaScript/other/filter.js:2]: @Sloth-Panda Please don't unresolve Comments if you have not actually resolved them. This defeats the purpose of putting a review and we obviously don't want to re-review such PRs.

File: JavaScript/other/filter.js (hunk 2 above)
Comment (HarshCasper) [reply JavaScript/other/filter.js:7]: What is the meaning of resolving conversations when you have not actually made the updates?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
> @HarshCasper @atarax665 can anyone check this its been 9-10 days now

Please make the changes as we suggested. Resolving the comments, without actually making the updates, defeats the purpose of having a PR review.

**Pull request closed without merging**